plugins/chat/app/services/chat/lookup_channel_threads.rb

```diff
@@ -54,28 +54,87 @@ def can_view_channel(guardian:, channel:, **)
     end
 
     def fetch_threads(guardian:, channel:, **)
-      Chat::Thread
-        .strict_loading
-        .includes(
-          :channel,
-          last_reply: %i[user uploads],
-          original_message_user: :user_status,
-          original_message: [
-            :chat_webhook_event,
-            :chat_mentions,
-            :chat_channel,
-            user: :user_status,
-          ],
+      threads =
+        Chat::Thread
+          .strict_loading
+          .includes(
+            :channel,
+            :user_chat_thread_memberships,
+            original_message_user: :user_status,
+            original_message: [
+              :chat_webhook_event,
+              :chat_mentions,
+              :chat_channel,
+              user: :user_status,
+            ],
+          )
+          .joins(
+            "JOIN (#{tracked_threads_subquery(guardian, channel)}) tracked_threads_subquery
+              ON tracked_threads_subquery.thread_id = chat_threads.id",
+          )
+          .joins(:user_chat_thread_memberships)
+          .where(user_chat_thread_memberships_chat_threads: { user_id: guardian.user.id })
+          .order(<<~SQL)
+            CASE WHEN user_chat_thread_memberships_chat_threads.last_read_message_id IS NULL
+              OR tracked_threads_subquery.latest_message_id >
+                user_chat_thread_memberships_chat_threads.last_read_message_id THEN 0 ELSE 1 END,
+              tracked_threads_subquery.latest_message_created_at DESC
+          SQL
+          .limit(50)
+          .to_a
+
+      last_replies =
+        Chat::Message
+          .strict_loading
+          .includes(:user, :uploads)
+          .from(<<~SQL)
+            (
+              SELECT thread_id, MAX(created_at) AS latest_created_at, MAX(id) AS latest_message_id
+              FROM chat_messages
+              WHERE thread_id IN (#{threads.pluck(:id).join(",")})
```

Comment:
```suggestion
              WHERE thread_id IN (#{threads.map(&:id).join(",")})
```

`threads` has already been loaded into memory at this point so there is no need to trigger another DB query with `pluck`.

Reply:
Ah good point thank you